pkg/ccl/changefeedccl/compression.go

```diff
@@ -17,24 +18,115 @@ import (
 	"github.com/klauspost/pgzip"
 )
 
-var useFastGzip = settings.RegisterBoolSetting(
-	settings.ApplicationLevel,
-	"changefeed.fast_gzip.enabled",
-	"use fast gzip implementation",
-	metamorphic.ConstantWithTestBool(
-		"changefeed.fast_gzip.enabled", true,
-	),
-	settings.WithPublic)
+var (
+	useFastGzip = settings.RegisterBoolSetting(
+		settings.ApplicationLevel,
+		"changefeed.fast_gzip.enabled",
+		"use fast gzip implementation",
+		metamorphic.ConstantWithTestBool(
+			"changefeed.fast_gzip.enabled", true,
+		),
+		settings.WithPublic)
+	gzipEncoderPool     = sync.Pool{}
+	fastGzipEncoderPool = sync.Pool{}
+	zstdEncoderPool     = sync.Pool{}
+	gzipDecoderPool     = sync.Pool{}
+	zstdDecoderPool     = sync.Pool{}
+)
 
 type compressionAlgo string
 
-const sinkCompressionGzip compressionAlgo = "gzip"
-const sinkCompressionZstd compressionAlgo = "zstd"
+const (
+	sinkCompressionGzip compressionAlgo = "gzip"
+	sinkCompressionZstd compressionAlgo = "zstd"
+)
 
 func (a compressionAlgo) enabled() bool {
 	return a != ""
 }
 
+type encoder interface {
+	io.WriteCloser
+	Reset(io.Writer)
+	Flush() error
+}
+
+type decoder interface {
+	io.ReadCloser
+	Reset(io.Reader) error
+}
+
+// encWrapper wraps an encoder with encoders pool.
+type encWrapper struct {
+	encoder encoder
+	pool    *sync.Pool
+}
+
+// Close flushes and closes the underlying encoder, resets it, and returns the wrapper to the pool.
+// This differs from the inner encoder's Close by adding pool management - the wrapper
+// is recycled after closing rather than being discarded.
+func (e *encWrapper) Close() error {
+	var err error
+	if fErr := e.encoder.Flush(); fErr != nil {
```

Comment:
Why do we need to call Flush here. Did we do it before? is Close not enough?

Reply:
You're right. There is no need to call flush since according to the docs close flushes unwritten data to reader. I'll remove it

Reply:
Fixed